### PR TITLE
Deprecate $mq-base-font-size setting

### LINKED ISSUE #122: Breakpoints at wrong width
I've ran into a issue and I'm not entirely sure what's causing it.

Currently building a website, where I have on the `_variables.scss` a `$base-font-size: 14pt`. This file is imported before the `_mq.scss` file.

In another file, I use it in the `html` selector to set a global font size, so that it gets inherited down.  Following that, and to keep things consistent, I've set the `$mq-base-font-size` on `_mq.scss` to be equal to `$base-font-size`.  Also added to new, bigger, breakpoints to the breakpoint list, but I don't think that has much impact on what's happening, since i've tested without them. 

I've noticed now that the breakpoints happen at the wrong widths of the viewport ( see attached ). Tried tracking down the root cause of the issue and it seems to lie somewhere on the  `mq-px2em` function. Removing it's call through the file, solves the issue and the breakpoints happen at the right widths. I do however lose some functionality, such as the use of the `$until`. 

image

### PR DESCRIPTION
**Pull request by kaelig**
Fixes a confusing part of the API, where the base font-size looked configurable, but it should never be 

Closes #122

### CODE REVIEW

**Comment by kaelig** [comment]
@sndrs this setting wasn’t documented in the README – does this look like a breaking change, or can we get away with a patch instead?

**Comment by sndrs** [comment]
cor i'm really not sure. 

i guess considering the following descriptions on semver.org:

> PATCH version when you make backwards-compatible bug fixes
> MINOR version when you add functionality in a backwards-compatible manner

could you call this backwards compatible? it could definitely break some installations

> MAJOR version when you make incompatible API changes

it's not part of the _documented_ API, but i think you could argue that's more of an accident in the docs/var naming? 

To my mind sass-mq invites user-land manipulation, in a way that say React does not. The setting wasn't documented but there was nothing to suggest it shouldn't be used and you could argue the `!default` sort of suggested it _could_ be used...

It does feel a bit weird to make it a major, given it's not a big refactor or something, but minor/patch feels like it would break the contract if you've used that var?

File: _mq.scss [hunk 1]
```diff
@@ -81,22 +77,23 @@ $mq-media-type: all !default;
...
 /// @example scss
 ///  $font-size-in-ems: mq-px2em(16px);
 ///  p { font-size: mq-px2em(16px); }
 ///
-/// @requires $mq-base-font-size
 /// @returns {Number}
-@function mq-px2em($px, $base-font-size: $mq-base-font-size) {
+@function mq-px2em($px) {
+    // em values always take 16px as a base font-size when in Media Queries
+    $base-font-size-for-em-media-queries: 16px;
```
Comment (sndrs) [inline _mq.scss]: would using `16px` as a magic number here be a valid use case? there's only 2 instances, and it kind of implies users shouldn't change it?

Comment (kaelig) [reply _mq.scss]: I think that from an idiomatic/declarative perspective, there’s a case for having it as a variable, but I’m not religiously attached to it…

**Comment by kaelig** [comment]
I agree with all the points you made about versioning.

Two things that would make me go for a patch rather than a major version:
1. I can’t imagine anyone overriding `$base-font-size` in their project, as it totally breaks the behaviour of sass-mq
2. It’s not part of the documented API

Do these arguments tip the jar enough for you in the favor of a patch as well?

**Comment by sndrs** [comment]
how about leaving things as they stand, but adding a few bits:

```scss
/// Base font size on the `<body>` element
/// @type Number (unit)
$mq-base-font-size: 16px; // do not change this or things may break!
```
and then

```scss
@function mq-px2em($px, $base-font-size: $mq-base-font-size) {
  @if $mq-base-font-size != 16px {
      @warn "Changing the value of $mq-base-font-size will probably break things!”;
  }
  /* etc */
}
```
this won’t break any installs with different values that are somehow working (¯\\_(ツ)_/¯), but should guide people away from potential problems in the future?

i think this change could then def be considered a patch?

**Comment by kaelig** [comment]
I love the idea, thank you @sndrs! I adapted the PR with various other fixes, as the sassdoc parsing had gone a bit wrong.

File: _mq.scss [hunk 2]
```diff
@@ -2,6 +2,12 @@
                   // See https://github.com/sass-mq/sass-mq/pull/10
 
 /// Base font size on the `<body>` element
+///
+/// Do not override this value, or things will break
+///
+/// @link https://github.com/sass-mq/sass-mq/issues/122
+/// @deprecated This setting will be removed in sass-mq v4
```
Comment (sndrs) [inline _mq.scss]: nice

Comment (kaelig) [reply _mq.scss]: Forgot we were already on v5…

I changed that line to say v6.0.0 instead (the next major version).

Comment (sndrs) [reply _mq.scss]: doh!

**Pull request merged**